sources/packages/backend/apps/api/src/route-controllers/atbc/atbc.students.controller.ts

```diff
@@ -45,18 +47,17 @@ export class ATBCStudentController extends BaseController {
     const student = await this.studentService.getStudentById(
       studentUserToken.studentId,
     );
-    // Check the PD status in DB. Student should only be allowed to check the PD status once.
-    // studentPDSentAt is set when student apply for PD status for the first.
-    // studentPDVerified is null before PD scheduled job update status.
-    // studentPDVerified is true if PD confirmed by ATBC or is true from sfas_individual table.
-    // studentPDVerified is false if PD denied by ATBC.
+    // To apply for a PD, SIN validation must be completed for the student and
+    // not applied for PD already.
     if (
       !student.sinValidation.isValidSIN ||
-      !!student.studentPDSentAt ||
-      student.studentPDVerified !== null
+      student.pdStatus !== PDStatus.NotRequested
     ) {
       throw new UnprocessableEntityException(
-        "Either the client does not have a validated SIN or the request was already sent to ATBC.",
+        new ApiProcessError(
+          "Either SIN validation is not complete or requested for PD already.",
```

Comment:
Is it not better to show separate error message for both the scenarios?

Reply:
@sh16011993 Sorry I do not follow. What are the error messages are you talking about?

Reply:
Resolving it as per the conversation with @dheepak-aot